src/Sarif/ExtensionMethods.cs

```diff
@@ -16,6 +16,40 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        public static string InstanceIdInstanceComponent(this RunAutomationDetails runAutomationDetails)
```

Comment:
>InstanceIdInstanceComponent [](start = 29, length = 27)

Consider: Make this an extension method on `string` so it would be used like this:
```
string instanceComponent = runAutomationDetails.InstanceId.InstanceComponent;
```
That seems more natural to me than
```
string instanceComponent = runAutomationDetails.InstanceIdInstanceComponent;
```
and it reduces the coupling of this method (you could unit test it without having the definition of `RunAutomationDetails`).

Reply:
Although I suppose you could argue that the current design eliminates the potential error of forgetting to null-check `runAutomationDetails.InstanceId`. Ok, I withdraw this.

---
In reply to: [226710524](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226710524) [](ancestors = 226710524)